media/js/src/chat.js

```diff
@@ -94,40 +68,46 @@ $(function() {
         entry.append('<div class="col-md-2 nick"><a href="' + data.userURL +
                      '">' + data.fullname + '</a></div>');
         var attr = 'message_text';
-        entry.append('<div class="col-md-9 ircmessage">' + data[attr] +
-                     '</div>');
+        entry.append('<div class="col-md-9 ircmessage">' +
+                     renderer.render(data[attr]) + '</div>');
         appendLog(entry);
     };
 
-    function appendLog(msg) {
+    var appendLog = function(msg) {
         var d = log[0];
         var doScroll = d.scrollTop === d.scrollHeight - d.clientHeight;
         msg.appendTo(log);
         if (doScroll) {
             d.scrollTop = d.scrollHeight - d.clientHeight;
         }
-    }
+    };
 
-    $('#msg_form').submit(function() {
-        var msg = $('#text-input');
-        $.ajax({
-            type: 'POST',
-            url: window.postURL,
-            data: {'text': msg.val()},
-            success: function() {
-                msg.val('');
-            },
-            error: function() {
-                alert('post failed');
-            }
+    domReady(function() {
+        if ($('#msg_form').length < 1) {
+            return;
+        }
```

Comment:
Yeah, I agree. I'll update this branch to use a separate `chat-built.js` on the chat page, like you've set up on the other branch.

We are doing this awkward "if ($('#msg_form').length < 1) {" thing in other places of the PMT javascript code as well:
https://github.com/ccnmtl/dmt/blob/master/media/js/src/views/item.js#L10
So yeah, we should be able to just have page-specific javascript that still takes advantage of requirejs's module loading behavior. Maybe the separate build for each custom module is the right approach.